packages/react-components/src/components/VideoGallery/useVideoStreamLifecycleMaintainer.ts

```diff
@@ -49,17 +50,44 @@ const useVideoStreamLifecycleMaintainer = (
     onDisposeStreamView,
     renderElementExists,
     scalingMode,
-    streamId
+    streamId,
+    /* @conditional-compile-remove(media-access) */
+    isVideoPermitted
   } = props;
 
   const [videoStreamViewResult, setVideoStreamViewResult] = useState<CreateVideoStreamViewResult | undefined>();
 
+  const createStreamViewTrampoline = useCallback(
+    (
+      isStreamAvailable?: boolean,
+      renderElementExists?: boolean,
+      /* @conditional-compile-remove(media-access) */ isVideoPermitted?: boolean
+    ) => {
+      /* @conditional-compile-remove(media-access) */
+      if (isStreamAvailable && !renderElementExists && isVideoPermitted) {
+        onCreateStreamView?.({ isMirrored, scalingMode })?.then((result) => {
+          result && setVideoStreamViewResult(result);
+        });
+      }
+
+      /* @conditional-compile-remove(media-access) */
+      return;
+
+      if (isStreamAvailable && !renderElementExists) {
+        onCreateStreamView?.({ isMirrored, scalingMode })?.then((result) => {
+          result && setVideoStreamViewResult(result);
+        });
+      }
+    },
+    [isMirrored, onCreateStreamView, scalingMode]
+  );
+
   useEffect(() => {
-    if (isStreamAvailable && !renderElementExists) {
```

Comment:
nit: You can just conditionally compile `isVideoPermitted` like so:
```
if (isStreamAvailable && !renderElementExists && /* @conditional-compile-remove(media-access) */ !renderElementExists ) {
...
}
```

Reply:
Nice :) will change it.